cinema/tests/test_movie_view_set.py

```diff
@@ -0,0 +1,201 @@
+
+from cinema.models import Movie, Genre, Actor
+from cinema.serializers import MovieSerializer, MovieListSerializer, MovieDetailSerializer
+
+MOVIE_URL = reverse("cinema:movie-list")
+
+def detail_url(movie_id):
+    return reverse("cinema:movie-detail", args=[movie_id])
+
+def sample_movie(**params) -> Movie:
+    defaults = {
+        "title": "The Great Brick",
+        "description": "The Great Brick fall on superman head",
+        "duration": 121
+    }
+    defaults.update(params)
+    return Movie.objects.create(**defaults)
+
+
+class TestUnauthenticatedUser(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+
+    def test_auth_required(self):
+        res = self.client.get(MOVIE_URL)
+        self.assertEqual(res.status_code, status.HTTP_401_UNAUTHORIZED)
+
+
+class TestAuthenticatedUser(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        self.user = get_user_model().objects.create_user(
+            email="user1@cinema.com",
+            password="2wsxcvfr4",
+        )
+        self.client.force_authenticate(self.user)
+
+        self.movie = sample_movie()
+        self.comedy_movie = sample_movie(title="The Great Brick 2")
+        self.drama_movie = sample_movie(title="The Great Brick 3")
+
+        self.genre_1 = Genre.objects.create(name="Comedy")
+        self.genre_2 = Genre.objects.create(name="Drama")
+
+        self.actor_1 = Actor.objects.create(
+            first_name="Adam",
+            last_name="Smith",
+        )
+        self.actor_2 = Actor.objects.create(
+            first_name="Eva",
+            last_name="Smith",
+        )
+
+        self.comedy_movie.genres.add(self.genre_1)
+        self.drama_movie.genres.add(self.genre_2)
+
+        self.comedy_movie.actors.add(self.actor_1)
+        self.drama_movie.actors.add(self.actor_2)
+
+        self.payload = {
+            "title": "The Great Brick 4",
+            "description": "The Great Brick fall on superman head",
+            "duration": 121,
+        }
+
+    def test_movie_list(self):
+        self.client.force_authenticate(self.user)
+        res = self.client.get(MOVIE_URL)
+        movies = Movie.objects.all()
+        serializer = MovieListSerializer(movies, many=True)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+
+        self.assertEqual(res.data, serializer.data)
+
+    def test_filter_movies_by_genres(self):
+        self.client.force_authenticate(self.user)
+        res = self.client.get(
+            MOVIE_URL, {"genres": f"{self.genre_1.id},{self.genre_2.id}"}
+        )
+
+        serializer = MovieListSerializer(self.movie)
+        serializer_with_genre_1 = MovieListSerializer(self.comedy_movie)
+        serializer_with_genre_2 = MovieListSerializer(self.drama_movie)
+
+        self.assertEqual(res.status_code, status.HTTP_200_OK)
+        self.assertNotIn(serializer.data, res.data)
+        self.assertIn(serializer_with_genre_1.data, res.data)
```

Comment:
This is a code quality issue. You're testing that the movie list can be filtered by genres. You're correctly checking that the response data includes the movies with the specified genres and excludes the movies without these genres.